bpf/bpf_sock.c

```diff
@@ -214,6 +214,12 @@ int sock4_xlate(struct bpf_sock_addr *ctx)
 
 	svc = __lb4_lookup_service(&key);
 	if (svc) {
+		// Do not perform service translation for external IPs because
+		// we don't want a k8s service to easily do MITM attacks for
+		// a public IP address.
+		if (svc->k8s_external) {
+			return CONNECT_PROCEED;
+		}
```

Comment:
No extra `{` `}` needed above. The UDP bits are missing below in `sock4_xlate_snd()`.

Is there a case where the externalIP service is overlapping with a nodeport service on that host? Afaik, it potentially could, so here we would hit the nodeport wildcard entry which seems fine. One thing that is a bit less clear to me is that depending on the order of insertion into the services map, we either override the IP:nodeport entry for the nodeport service with the external IP one or vice versa?